src/pybind/mgr/volumes/fs/operations/volume.py

```diff
@@ -135,10 +145,34 @@ def delete_volume(mgr, volname, metadata_pool, data_pools):
     result_str = "metadata pool: {0} data pool: {1} removed".format(metadata_pool, str(data_pools))
     return r, result_str, ""
 
+def _perform_checks_before_renaming_vol(mgr, volname):
+    mdsmap = get_mds_map(mgr, volname)
+
+    if mdsmap is None:
+        return 0, ''
+
+    if len(mdsmap['up']) > 0:
+        return (-errno.EPERM,
+                (f'Volume "{volname}" is not offline. Before renaming a '
+                  'volume, it must be marked as down. See `ceph fs fail`.'))
+
+    if not is_flag_set(mdsmap, CEPH_MDSMAP_REFUSE_CLIENT_SESSION):
+        return (-errno.EPERM,
+                (f'Volume "{volname}" doesn\'t refuse clients. Before '
+                  'renaming a volume, flag "refuse_client_session" must be '
+                  'set. See `ceph fs fail`.'))
+
+    return 0, ''
+
 def rename_volume(mgr, volname: str, newvolname: str) -> Tuple[int, str, str]:
     """
     rename volume (orch MDS service, file system, pools)
     """
+    errno, errmsg = _perform_checks_before_renaming_vol(mgr, volname)
```

Comment:
Is there a need for this check here since the check (involving `CEPH_MDSMAP_REFUSE_CLIENT_SESSION`) is done in the monitor?